src/client/producer/broadcast.rs

```diff
@@ -81,35 +106,26 @@ mod tests {
 
         // Test simple receiver
         let broadcast: BroadcastOnce<usize> = Default::default();
-        let receiver = broadcast.receive();
+        let receiver = broadcast.receiver();
```

Comment:
I know this wasn't here before, but could we have a simple "is pending" check as well as `peek` before broadcasting, e.g. 

```suggestion
        let receiver = broadcast.receiver();
        assert!(receiver.peek().is_none());
        assert_is_pending(receiver.receive()).await;
```